patches/tModLoader/Terraria/GameContent/ShimmerTransforms.cs.patch

```diff
@@ -0,0 +1,54 @@
+--- src/TerrariaNetCore/Terraria/GameContent/ShimmerTransforms.cs
++++ src/tModLoader/Terraria/GameContent/ShimmerTransforms.cs
+@@ -1,4 +_,5 @@
+ using Terraria.ID;
++using Terraria.ModLoader;
+ 
+ namespace Terraria.GameContent;
+ 
+@@ -12,8 +_,15 @@
+ 
+ 	public static int GetDecraftingRecipeIndex(int type)
+ 	{
++		/*
++		vanilla:
+ 		int num = ItemID.Sets.IsCrafted[type];
+ 		if (num < 0)
++		return false;
++		*/
++
++		int[] recipeIndecies = ItemID.Sets.IsCrafted[type];
++		if (recipeIndecies[0] < 0)
+ 			return -1;
+ 
+ 		if (WorldGen.crimson && ItemID.Sets.IsCraftedCrimson[type] >= 0)
+@@ -22,7 +_,18 @@
+ 		if (!WorldGen.crimson && ItemID.Sets.IsCraftedCorruption[type] >= 0)
+ 			return ItemID.Sets.IsCraftedCorruption[type];
+ 
++		//Deliberately preserving the "last first" order of the previous implementation, adding manual priority support
++		int recipe = recipeIndecies[recipeIndecies.Length -1];
++
++		foreach (int posRec in recipeIndecies) {
++			if (RecipeLoader.DecraftAvailable(Main.recipe[posRec])) {
++				if (Main.recipe[posRec].shimmerPriority >= Main.recipe[recipe].shimmerPriority)
++					recipe = posRec;
++
++			}
++		}
++
+-		return num;
++		return recipe;
+ 	}
```

Comment:
for patch reduction, change `recipe` to `num`, and get rid of the empty line after the last `posRec` reference

Reply:
👍 Done the list + some other stuff